src/test/java/com/arpnetworking/metrics/common/kafka/ConsumerDeserializerTest.java

```diff
@@ -0,0 +1,143 @@
+            + "\"group.id\":\"test\",\"client.id\":\"consumer0\","
+            + "\"key.deserializer\":\"org.apache.kafka.common.serialization.StringDeserializer\","
+            + "\"auto.offset.reset\":\"earliest\""
+            + "},\"topics\":[\"" + TOPIC + "\"]}";
+
+    private static final String CONSUMER_JSON_NO_BOOTSTRAP_SERVER = "{\"configs\": {"
+            + "\"group.id\":\"test\",\"client.id\":\"consumer0\","
+            + "\"key.deserializer\":\"org.apache.kafka.common.serialization.StringDeserializer\","
+            + "\"value.deserializer\":\"org.apache.kafka.common.serialization.StringDeserializer\","
+            + "\"auto.offset.reset\":\"earliest\""
+            + "},\"topics\":[\"" + TOPIC + "\"]}";
+
+    private static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.createInstance();
+
+    @BeforeClass
+    public static void setUpClass() throws IOException {
+        Files.createDirectories(new File("./target/tmp/filter/ConsumerDeserializer").toPath());
+        final SimpleModule module = new SimpleModule("KafkaConsumer");
+        module.addDeserializer(Consumer.class, new ConsumerDeserializer<>());
+        OBJECT_MAPPER.registerModule(module);
+    }
+
+    @Test
+    public void testDeserializeConsumerSuccess() throws IOException {
+        final Consumer<String, String> consumer = createConsumerFromJSON(CONSUMER_JSON_VALID, "validConsumer.json");
+        Assert.assertEquals(Sets.newHashSet(TOPIC), consumer.subscription());
+    }
+
+    @Test(expected = JsonParseException.class)
+    public void testDeserializeConsumerMalformedJson() throws IOException {
+        createConsumerFromJSON(CONSUMER_JSON_MALFORMED, "malformedJson.json");
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testDeserializeConsumerMissingKeyDeserializer() throws IOException {
+        createConsumerFromJSON(CONSUMER_JSON_NO_KEY_DESERIALIZER, "keyDeserializerMissing.json");
+    }
+
+    @Test(expected = ConfigException.class)
+    public void testDeserializeConsumerMissingValueDeserializer() throws IOException {
+        createConsumerFromJSON(CONSUMER_JSON_NO_VALUE_DESERIALIZER, "valueDeserializerMissing.json");
+    }
+
+    @Test(expected = KafkaException.class)
+    public void testDeserializeConsumerMissingBootstrapServer() throws IOException {
+        createConsumerFromJSON(CONSUMER_JSON_NO_BOOTSTRAP_SERVER, "bootstrapServerMissing.json");
+    }
+
+    private static Consumer<String, String> createConsumerFromJSON(final String jsonString, final String fileName)
+            throws IOException {
+        final File file = new File("./target/tmp/filter/ConsumerDeserializer/" + fileName);
+        Files.write(file.toPath(), jsonString.getBytes(Charsets.UTF_8));
+
+
+        final Consumer<String, String> consumer = OBJECT_MAPPER.readValue(
+                new File("./target/tmp/filter/ConsumerDeserializer/" + fileName),
+                new TypeReference<Consumer<String, String>>() {});
```

Comment:
I believe you can simply use `OBJECT_MAPPER.readValue(jsonString, new TypeReference...)`, and not need to muck about with files!